CS/VehicleDekho/src/com/vehicledekho/core/helper/VehicleHelper.java

```diff
@@ -0,0 +1,46 @@
+package com.vehicledekho.core.helper;
+
+import com.vehicledekho.core.Bike;
+import com.vehicledekho.core.Car;
+import com.vehicledekho.core.Vehicle;
+
+public class VehicleHelper {
+	public static Vehicle createVehicle(String vehicleType, String model,
+			String make) {
+		if (vehicleType.equalsIgnoreCase("car")) {
+			return createCar(model, make);
+		} else {
+			return carBike(model, make);
+		}
+	}
+
+	private static Vehicle carBike(String model, String make) {
+		Bike bike = new Bike();
+		bike.setModel(model);
+		bike.setMake(make);
+		bike.setSelfStart(true);
```

Comment:
here you only create the bike of one type only with same features. What about the bikes with different features? You need to create a method in Bike class which take the parameter like selfStart, helmetprice and so on. Due to this you are able to create bike with different features.